Sources/Bag.swift

```diff
@@ -22,11 +41,22 @@ public struct Bag<Element> {
 	/// - parameters:
 	///   - value: A value that will be inserted.
 	@discardableResult
-	public mutating func insert(_ value: Element) -> RemovalToken {
-		let token = RemovalToken()
-		let element = BagElement(value: value, token: token)
+	public mutating func insert(_ value: Element) -> Token {
+		var timestamp: UInt64 = 0
+
+		repeat {
+			// `DispatchTime` is expected to use monotonic clocks. That said a guard is
+			// put here just in case.
+			timestamp = DispatchTime.now().uptimeNanoseconds
+		} while timestamp <= lastTimestamp
+
+		lastTimestamp = timestamp
+
+		let token = Token(timestamp)
```

Comment:
`Bag` was using a monotonic counter before. But overflowing seems to be a concern, so the identifier was stored inside a referenced token to allow reindexing as the overflow happens. Then after a while, we have stripped it since relying on the object identity is enough if an token object is used.

The point of a timestamp here is that the monotonic clock is (sadly?) guaranteed not to overflow.